wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -947,22 +948,15 @@ class PlatformRepo @Inject constructor(
         } else {
             authenticationChain
         }
-        val key = decryptedChain.getKey(index) // watchingKey
+        val key = decryptedChain.getKey(index)
         Preconditions.checkState(key.path.last().isHardened)
         return key
 
     }
 
     fun getIdentityFromPublicKeyId(): Identity? {
         val encryptionKey = getWalletEncryptionKey()
-        val firstIdentityKey = try {
-            getBlockchainIdentityKey(0, encryptionKey) ?: return null
-        } catch (e: KeyCrypterException.InvalidCipherText) {
-            log.info("failure to decrypt identity keychain", e)
-            log.info("attempt again to obtain the wallet encryption key and the identity")
-            val encryptionKeyTwo = getWalletEncryptionKey()
-            getBlockchainIdentityKey(0, encryptionKeyTwo) ?: return null
-        }
+        val firstIdentityKey = getBlockchainIdentityKey(0, encryptionKey) ?: return null
```

Comment:
The try/catch was intended to fix a decryption error, but that error wasn't fixed by retrying in the catch block.